lib/_stream_transform.js

```diff
@@ -217,3 +229,109 @@ function done(stream, er, data) {
     throw new ERR_TRANSFORM_ALREADY_TRANSFORMING();
   return stream.push(null);
 }
+
+function SourceIterator(asyncGeneratorFn, opts) {
+  const source = this;
+  this[kSourceIteratorPull] = null;
```

Comment:
I was thinking of using the new private syntax (e.g. `this.#pull = null`) but decided against since it will make backporting to 10 (and maintenance) more challenging

Reply:
+1